src/3d/symbols/qgspointcloud3dsymbol_p.cpp

```diff
@@ -225,6 +226,36 @@ void QgsPointCloud3DSymbolHandler::makeEntity( Qt3DCore::QEntity *parent, const
   // cppcheck-suppress memleak
 }
 
+QgsPointCloudBlock *QgsPointCloud3DSymbolHandler::pointCloudBlock( QgsPointCloudIndex *pc, const IndexedPointCloudNode &n, const QgsPointCloudRequest &request, const QgsPointCloud3DRenderContext &context )
+{
+  QgsPointCloudBlock *block = nullptr;
+  if ( pc->accessType() == QgsPointCloudIndex::AccessType::Local )
+  {
+    block = pc->nodeData( n, request );
+  }
+  else if ( pc->accessType() == QgsPointCloudIndex::AccessType::Remote )
+  {
+    bool loopAborted = false;
+    QEventLoop loop;
+    QgsPointCloudBlockRequest *req = pc->asyncNodeData( n, request );
+    QObject::connect( req, &QgsPointCloudBlockRequest::finished, &loop, &QEventLoop::quit );
+
+    QMetaObject::Connection connection = QObject::connect( &context, &QgsPointCloud3DRenderContext::renderingCanceled, [ & ]()
+    {
+      loopAborted = true;
+      loop.quit();
+    } );
+    loop.exec();
+    QObject::disconnect( connection );
```

Comment:
It should not be necessary to disconnect from the connection like this... at least it's not something I would have seen in QGIS code before. To make sure the connection is disconnected when `loop` is deleted, you can use QObject::connect() variant where you can give it `const QObject *context` object (`&loop` in this case).

Reply:
I didn't know that existed - Thanks